src/libraries/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/CertificateRequestApiTests.cs

```diff
@@ -179,32 +181,33 @@ public static void CtorValidation_RSA_X500DN()
         {
             X500DistinguishedName subjectName = null;
             RSA key = null;
-            HashAlgorithmName hashAlgorithm = default(HashAlgorithmName);
             RSASignaturePadding padding = null;
 
             AssertExtensions.Throws<ArgumentNullException>(
                 "subjectName",
-                () => new CertificateRequest(subjectName, key, hashAlgorithm, padding));
+                () => new CertificateRequest(subjectName, key, default(HashAlgorithmName), padding));
 
             subjectName = new X500DistinguishedName("");
 
             AssertExtensions.Throws<ArgumentNullException>(
                 "key",
-                () => new CertificateRequest(subjectName, key, hashAlgorithm, padding));
+                () => new CertificateRequest(subjectName, key, default(HashAlgorithmName), padding));
 
             key = RSA.Create(TestData.RsaBigExponentParams);
 
             using (key)
             {
-                AssertExtensions.Throws<ArgumentException>(
+                AssertExtensions.Throws<ArgumentNullException>(
                     "hashAlgorithm",
-                    () => new CertificateRequest(subjectName, key, hashAlgorithm, padding));
+                    () => new CertificateRequest(subjectName, key, default(HashAlgorithmName), padding));
 
-                hashAlgorithm = HashAlgorithmName.SHA256;
+                AssertExtensions.Throws<ArgumentException>(
+                    "hashAlgorithm",
+                    () => new CertificateRequest(subjectName, key, new HashAlgorithmName(""), padding));
 
                 AssertExtensions.Throws<ArgumentNullException>(
                     "padding",
-                    () => new CertificateRequest(subjectName, key, hashAlgorithm, padding));
+                    () => new CertificateRequest(subjectName, key, HashAlgorithmName.SHA256, padding));
```

Comment:
nit: it has been already like that before your changes, but I think that the readability could by improved by using `null` in explicit way for all the `null` arguments rather than having a local variable with `null` value

```suggestion
                    () => new CertificateRequest(subjectName, key, HashAlgorithmName.SHA256, padding: null));
```

Reply:
I agree.  But I also want to respect the preferences of the code owner here.